packages/react/src/Card/Card.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {forwardRef, useCallback} from 'react'
+import {isFragment} from 'react-is'
+import clsx from 'clsx'
+
+import {Heading, HeadingTags, Text} from '..'
+import {ExpandableArrow} from '../ExpandableArrow'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/card/base.css'
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/card/colors-with-modes.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './Card.module.css'
+import stylesLink from '../Link/Link.module.css'
+
+export type CardProps = {
+  /**
+   * Valid children include Card.Image, Card.Heading, and Card.Description
+   */
+  children: React.ReactNode | React.ReactElement<CardHeadingProps> | React.ReactElement<CardDescriptionProps>
+  /**
+   * Changes the link label of the card
+   * * @default 'Learn more'
+   * */
+  label?: string
+  /**
+   * The href of the link
+   * */
+  href: string
+  /**
+   * Event handlers
+   * */
+  onMouseEnter: React.MouseEventHandler<HTMLAnchorElement>
+  onMouseLeave: React.MouseEventHandler<HTMLAnchorElement>
+  onFocus: React.FocusEventHandler<HTMLAnchorElement>
+  onBlur: React.FocusEventHandler<HTMLAnchorElement>
+} & BaseProps<HTMLAnchorElement> &
+  React.ComponentPropsWithoutRef<'a'>
+
+const CardRoot = forwardRef<HTMLAnchorElement, CardProps>(
+  ({onMouseEnter, onMouseLeave, onFocus, onBlur, children, className, label, href, ...props}) => {
+    const [isHovered, setIsHovered] = React.useState(false)
+    const [isFocused, setIsFocused] = React.useState(false)
+
+    const handleMouseEnter = useCallback(
+      (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+        setIsHovered(!isHovered)
+        onMouseEnter(event)
+      },
+      [onMouseEnter, isHovered]
+    )
+
+    const handleMouseLeave = useCallback(
+      (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+        setIsHovered(!isHovered)
+        onMouseLeave(event)
+      },
+      [onMouseLeave, isHovered]
+    )
+
+    const handleOnFocus = useCallback(
+      (event: React.FocusEvent<HTMLAnchorElement, Element>) => {
+        setIsFocused(!isFocused)
+        onFocus(event)
+      },
+      [onFocus, isFocused]
+    )
+
+    const handleOnBlur = useCallback(
+      (event: React.FocusEvent<HTMLAnchorElement, Element>) => {
+        setIsFocused(!isFocused)
+        onBlur(event)
+      },
+      [onBlur, isFocused]
+    )
+
+    const childrenHasFragment = React.Children.toArray(children).some(child => isFragment(child))
+    const filteredChildren = React.Children.toArray(children).filter(child => {
+      if (React.isValidElement(child) && typeof child.type !== 'string') {
+        if (childrenHasFragment || child.type === CardHeading || child.type === CardDescription) {
```

Comment:
To optimise this a little, could we call `isFragment(child)` here directly? It'd avoid having to create `childrenHasFragment` reference using a separate loop as we only call it once. We can update FAQ to do the same also?

```suggestion
        if (isFragment(child) || child.type === CardHeading || child.type === CardDescription) {
```